### DISCUSSION TITLE
PAID: Using ImageMagick to go from an .XLS file to a .PSD

### DISCUSSION THREAD

**Discussion by nickhart121** [ImageMagick/ImageMagick#3704] [Consulting]
I am not a programmer and am not familiar with ImageMagick. I need to be able to go from an Excel spreadsheet containing text using Arial Unicode (or another font I define) and output each cell to an Adobe Photoshop .PSD file with a layer containing the text, still in an editable format. 

The .PSD will be a 300 DPI, rectangular image file 370 px wide x 90 px tall. Each block of text must be maximized to fill the entire rectangular area. From there, I also need a command line to convert .PSD to .BMP, .PNG or other graphic file formats. 

Ideally, I also need a few other functions if possible:

File output naming/numbering control
The ability for ImageMagick to detect an excessive amount of text, and switch to another font like Arial Narrow, if necessary
The ability for ImageMagick to automatically use larger point sizes for small amounts of text, so that the text fills out the entire rectangular area

Thanks in advance! If you can help, please let me know so we can discuss further.

**Comment by snibgo** [comment]
> ... and output each cell to an Adobe Photoshop .PSD file with a layer containing the text, still in an editable format.

ImageMagic is a raster image processor. It cannot put text into a PSD, with that text editable.

**Reply by nickhart121** [reply]
Thanks for the response. So ignoring the editable text, does the rest of what I'm talking about seem doable with ImageMagick?

**Comment by snibgo** [comment]
Yes, in a shell script. It could work something like this:

1. From inside Excel, save the spreadsheet as a CSV file (ie text, with commas separating the values).
2. A shell script splits out each cell value as text.
3. For each cell text, IM rasterizes it (using "caption:"), choosing the largest point size that fits a 370x90 pixel box. If the pointsize is too small (ie less than your pre-defined limit), try again with a thinner font.

From that, the result would be a load of images. These could be saved in separate files, or a single PSD, or whatever.

I'm not available to do this. Sorry.

**Reply by nickhart121** [reply]
Thanks, I appreciate the help!
